packages/app-project/src/shared/components/CollectionsModal/CollectionsModalContainer.js

```diff
@@ -1,125 +1,99 @@
-import { inject, observer } from 'mobx-react'
-import { array, func } from 'prop-types'
-import { Component } from 'react'
+import { MobXProviderContext, observer } from 'mobx-react'
+import { useContext, useState } from 'react'
+import { bool, func, string } from 'prop-types'
 
 import CollectionsModal from './CollectionsModal'
 import SelectCollection from './components/SelectCollection'
 import CreateCollection from './components/CreateCollection'
 
-function storeMapper (stores) {
-  const store = stores.store || {}
-  const collectionsStore = store.user?.collections || {}
-  const {
-    addSubjects,
-    collections,
-    createCollection,
-    searchCollections
-  } = collectionsStore
+function useStore() {
+  const { store } = useContext(MobXProviderContext)
+
+  const { addSubjects, collections, createCollection } = store.user?.collections
+
   return {
     addSubjects,
     collections,
-    createCollection,
-    searchCollections
+    createCollection
   }
 }
 
-class CollectionsModalContainer extends Component {
-  constructor () {
-    super()
-    this.addToCollection = this.addToCollection.bind(this)
-    this.close = this.close.bind(this)
-    this.createCollection = this.createCollection.bind(this)
-    this.onSelect = this.onSelect.bind(this)
-    this.updateCollection = this.updateCollection.bind(this)
-    this.state = {
-      active: false,
-      newCollection: {
-        display_name: '',
-        private: false
-      },
-      selectedCollection: undefined,
-      subjectId: null
-    }
-  }
+const DEFAULT_HANDLER = () => true
+const defaultNewCollection = {
+  display_name: '',
+  private: false
+}
+
+const CollectionsModalContainer = ({
+  collectionsModalActive = false,
+  setCollectionsModalActive = DEFAULT_HANDLER,
+  subjectID = ''
+}) => {
+  const {
+    addSubjects = DEFAULT_HANDLER,
+    collections = [],
+    createCollection = DEFAULT_HANDLER
+  } = useStore()
+
+  const [newCollection, setNewCollection] = useState(defaultNewCollection)
+  const [selectedCollection, setSelectedCollection] = useState(undefined)
+  const [selectedSubjectID, setSelectedSubjectID] = useState(subjectID)
 
-  addToCollection (event) {
+  const addToCollection = (event) => {
```

Comment:
There's no `this` context in a functional component so you probably don't need arrows to bind the calling context.